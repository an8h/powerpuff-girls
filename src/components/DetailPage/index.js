// @flow

/* 
DetailPage Component: 
It is shown when the user clicks an episode on Home page. 
Rendered UI:
item   : Shows a selected episode
*/

import React, { useState } from "react";
import { Button } from "reactstrap";
import { Redirect } from "react-router-dom";
import ItemToShow from "../ItemToShow/index";

const DetailPage = (props: Object) => {
  const { location } = props;
  const { state } = location;
  const { clickedItem } = state;
  const [closeButtonClicked, handleClick] = useState(false);

  return (
    <div>
      <ItemToShow item={clickedItem} />
      <Button
        className="button"
        close
        onClick={() => {
          handleClick(true);
        }}
      />
      {closeButtonClicked ? <Redirect to="/" /> : null}
    </div>
  );
};

export default DetailPage;
