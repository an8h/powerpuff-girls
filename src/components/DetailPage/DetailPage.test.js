import React from "react";
import renderer from "react-test-renderer";
import DetailPage from "./index";

const props = {
  location: {
    pathname: "/detail",
    search: "",
    hash: "",
    key: "",
    state: {
      clicked: true,
      clickedItem: {
        name: "this is a clicked item",
        summary: "This is a test summary",
        image: {
          medium: "",
          original: "",
        },
      },
    },
  },
};

describe("DetailPage Component", () => {
  let component;

  beforeEach(() => {
    component = renderer.create(
      <DetailPage
        location={props.location}
        clickedItem={props.location.state.clickedItem}
      />
    );
  });

  it("should render correctly", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
