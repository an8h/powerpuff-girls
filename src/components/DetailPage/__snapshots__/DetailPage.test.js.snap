// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage Component should render correctly 1`] = `
<div>
  <div>
    <div
      className="container"
    >
      <img
        alt="this is a clicked item"
        className="image"
        onError={[Function]}
        src=""
      />
      <div
        className="text"
      >
        <h2
          className="header"
        >
          this is a clicked item
        </h2>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "This is a test summary",
            }
          }
        />
      </div>
    </div>
  </div>
  <button
    aria-label="Close"
    className="button close"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden={true}
    >
      Ã—
    </span>
  </button>
</div>
`;
